### DISCUSSION TITLE
hasAnyText should trim white space?

### DISCUSSION THREAD

**Discussion by simonihmig** [simplabs/qunit-dom#1419] [Q&A]
`hasAnyText` returns true even if `textContent`contains only white space (spaces and/or line breaks). Is this expected?

For something like this:

```hsb
<p data-test-message>
{{#if this.showMessage}}
  {{this.message}}
{{/if}}
</p>
```

`assert.dom('[data-test-message]').hasAnyText()` would *always* return true!

**Comment by Turbo87** [comment]
hmm... I guess in that case the whitespace collapsing would collapse it down to a single whitespace, but the content would still be `' '`, not `''`. so I guess maybe that is indeed correct? 😅

**Reply by simonihmig** [reply]
> the whitespace collapsing would collapse it down to a single whitespace

Why that?

![image](https://user-images.githubusercontent.com/1325249/149793000-f315a75e-57de-4b41-9505-8f1c26c81ffb.png)

**Reply by simonihmig** [reply]
And however it is implemented, the question is how it is designed? Shouldn't a whitespace-only string be considered "empty", at least in the context of `hasAnyText`? See my real world example above...

**Reply by Turbo87** [reply]
it's using https://github.com/simplabs/qunit-dom/blob/master/lib/helpers/collapse-whitespace.ts, though I don't remember exactly where I've found the rules for that algorithm 😅 

it is meant to somewhat resemble https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace
